### PR TITLE
Update option to use style config

### PR DESCRIPTION
**Pull request by colinlohner**
**Overview**
Updated `Option` to use style config

**Screenshots (if applicable)**
Screen Recording 2020-09-29 at 12 03 40 PM

**Documentation**
- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/themes/default/components/option.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  },
+
+  _active: {
+    backgroundColor: 'intents.info.background',
+  },
+
+  _selected: {
+    backgroundColor: 'intents.info.background',
+
+    ' span': {
```
Comment (mshwery) [inline src/themes/default/components/option.js:42]: Should this be `& span`? IIRC it will work in glamor, but think we should be consistent.

Comment (mshwery) [reply src/themes/default/components/option.js:42]: Also not a huge fan that we have knowledge of the component architecture in the theme.

File: src/select-menu/src/Option.js [hunk 2]
```diff
@@ -22,45 +41,31 @@ const Option = memo(
...
-    }
+    const { className: themedClassName, ...boxProps } = useStyleConfig(
+      'Option',
+      emptyObject,
+      pseudoSelectors,
+      internalStyles
+    )
 
     return (
       <TableRow
```
Comment (akleiner2) [inline src/select-menu/src/Option.js:52]: Is it worth breaking apart from `<TableRow />` or is that going to come in a follow-up PR?

File: src/select-menu/src/Option.js [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+  alignItems: 'center',
+  display: 'flex'
+}
+
+const emptyObject = {}
 
 const Option = memo(
   forwardRef(function Option(props, ref) {
     const {
+      className,
```
Comment (akleiner2) [inline src/select-menu/src/Option.js:30]: Do we need to add the additional classname here? I don't think we export `<Option />` in EG, so consumers can't style this initially.

**Review by akleiner2 — APPROVED** [review]
Non-blocking comments to consider!

**Comment by akleiner2** [comment]
Also - unsure why Chromatic is failing 🤔

**Review by mshwery — APPROVED** [review]

File: src/select-menu/src/Option.js (hunk 2 above)
Comment (colinlohner) [reply src/select-menu/src/Option.js:52]: In a follow up

File: src/themes/default/components/option.js (hunk 1 above)
Comment (colinlohner) [reply src/themes/default/components/option.js:42]: Agreed, we will change this in the follow up pr

File: src/select-menu/src/Option.js (hunk 3 above)
Comment (colinlohner) [reply src/select-menu/src/Option.js:30]: that makes sense, i will remove this

**Pull request merged**